README.md

```diff
@@ -1,15 +1,142 @@
+
+## Auth Component:
+
+Auth component handles the authentication and authorization of users. It is responsible for managing user sessions, verifying user credentials, and granting access to protected resources based on the user's role and permissions.
+
+The auth component typically includes a login as well as a registration form for new users to create an account. The component may also handle password reset functionality and provide options for users to manage their accounts.
+
+# Core Component:
+
+## API:
+
+- Adapters:We are using 3 adapters in this microservice named "anyAdapter,countAdapter,nameAdapter".We use these adapters for manipulating the data, as we get the data from the api.
+
+- Commands:In this microservice we are using "del-api.command,get-api.command,get-list-api.command,patch-api.command,post-api.command,put-api.command" We are using all the commands for
+  modifying the data through API.
+
+- Models:We are using 3 models named "count.model,named-id-required.model,named-id.model" In these models we are using model based validations.
+
+## Auth:
+
+- Adaptors:Two adapter service provided are "logged-in-user-adapter.service and login-adapter.
+  service" which helps in adapting API response to a LoggedInUserDM model representing a logged-in user or adapting the LoginModel object to the API request format and adapting the API response which allows for a separation of concerns between the API response data and the application data model.
+
+- Commands:It provides implementation of a commands that makes a Get/Post/patch request to the
+  API endpoint for triggering forget password functionality,retrieving current user's data,authentication token,initiating a Google OAuth login flow and logout etc.
+
+- Guards:In microservice 2 guards provided are "LoggedInGuard and AuthGuard" used to control
+  access to certain routes or pages in the application.Guards can be used to check if the user has permission to access a certain route, if the user is logged in, or to perform other types of authentication or authorization check.
+
+- Models: We are using 4 models named "logged-in-user.model,loggedin-response.model,login.model,
+  token-response.model" that represents user's data when they are logged in,response from a login API call(code),data that is required for a user to authenticate or log
+  in to a system and the the response from server after a user gets authenticated
+  and obtains a new access token and refresh token.
+
+## Decorator:
+
+In this microservice we are using 2 decorators named "required.decorator,validate.decorator",In these decorator we are providing our requirement for the data that which data needs to be fill,and than validate the data.
```

Comment:
Fix the grammar, formatting and spelling mistakes.

Reply:
Done